### PR TITLE
fix question mark issue

### PR DESCRIPTION
**Pull request by Logan676**
@forouher @lins05 the question mark "?" issue was fixed in the [new web API](https://github.com/haiwen/seafile-docs/blob/master/develop/web_api.md#get-thumbnail-image)

### CODE REVIEW

**Comment by shoeper** [comment]
It'll break the thumbnail functionality if the server isn't completely up to date, won't it?

I'm not sure if the app has the server version somwhere but I'd recommend to include a workaround for a few month (e.g. 3).

**Comment by forouher** [comment]
If the server doesn't support the (new) thumbnail API, it should return an error code. So Seadroid  can only display a thumbnail if it has a cached copy of the image. Which is exactly the current behaviour of Seadroid.

So there shouldn't be any regression.

**Comment by shoeper** [comment]
As far as I can see the api has been slightly changed. So servers running 4.x but not the upcoming 4.y release won't be able to serve thumnails because the `?`is not supported by them.

**Comment by forouher** [comment]
Yes. 4.0.x won't be able to serve thumbnails just as the 3.x and 2.x servers couldn't. So basically the thumbnail feature will only work with 4.1 upwards.

But keep in mind that up to now there was no Seadroid release that supports server-side thumbnails. This is a patch for a feature that is (as of now) still unreleased.

I guess one could add a specific workaround for 4.0.x servers. But that seems like overkill for an open source software with relatively short release cycles (and with such a simple to use upgrade mechanism :).

File: src/com/seafile/seadroid2/data/DataManager.java [hunk 1]
```diff
@@ -143,8 +143,7 @@ public String getThumbnailLink(String repoName, String repoID, String filePath,
         } else {
             try {
                 String pathEnc = encodeThumbnailFilePath(filePath);
-                // TODO: If there is a "?" in the path, this will break
-                return account.getServer() + String.format("api2/repos/%s/thumbnail/%s?s=%s", repoID, pathEnc, size);
+                return account.getServer() + String.format("api2/repos/%s/thumbnail/?p=%s&size=%s", repoID, pathEnc, size);
```
Comment (lins05) [inline src/com/seafile/seadroid2/data/DataManager.java]: The hacky escaping of the path param in [DataManager.encodeThumbnailFilePath](https://github.com/haiwen/seadroid/tree/5a7531da03e2058c02b15a9ca4f21b459630d540/src/com/seafile/seadroid2/data/DataManager.java#L126) is no longer necessary. We can replace it with a call to `URLEncoder.encode()`.

**Pull request merged**